### DISCUSSION TITLE
tarpc

### DISCUSSION THREAD

**Discussion by thomasza92** [c410-f3r/wtx#287] [General]
Hello! First I want to say that I've found this project to be super interesting and really useful for comparing all the different RPC frameworks. Thanks for all the hard work!

I was wondering if anyone had any thoughts on [tarpc](https://github.com/google/tarpc)? 

Is it good? Is it fast? I would love to know! 

I also find it at least somewhat curious that there is a working example for gRPC but not for tarpc since tarpc is written in Rust! 

I wouldn't mind taking a whack at trying to implement this myself but I'm a bit green when it comes to working with RPCs in general so I'm not sure how far I'd get!

Thanks again and hopefully someone doesn't mind to enlighten me a bit 😁

**Comment by c410-f3r** [comment]
Hello,

Unfortunately the internals of `tarpc` are unknown to me. The project appears to be API framework similar to what `wtx` already provides, i.e., endpoint schemas defined in the code using procedural macros. If so, then `wtx` is more like a competitor of `tarpc`.

`wtx` favors flexibility and performance over simplicity, as such, the project is quite complex. For what it is worth, I recommend contributions to more well established repositories like the ones listed at https://goodfirstissue.dev/language/rust.

Cheers
